RocketChat/Rocket.Chat
# fix: Inactive Clients Not Refreshing Action Buttons After App Deployment

## Proposed changes (including videos or screenshots)

Currently, when the app is redeployed, active clients properly fetch and display the updated action buttons (via GET /apps/actionButtons). However, any clients that were inactive during the deployment (e.g., a laptop in sleep mode) do not fetch the updated list upon becoming active again. As a result, these inactive clients continue using outdated actionIds, causing inconsistent and potentially broken interactions.

So now, everytime the client goes offline, we invalidate the cache, since the buttons might have changed in the meantime

## Issue(s)


## Steps to test or reproduce


Start with the client active and confirm it fetches updated action buttons upon deployment.
Put the client into an inactive state (e.g., close laptop lid or simulate a network disconnection), then redeploy the app.
Reactivate the client and verify that it immediately retrieves the updated action buttons list.
Repeat steps 2-3 for multiple clients to ensure consistent behavior across different endpoints and environments.

Note: You can use `Meteor.disconnect` to test socket reconnection

## Further comments

[CONN-434](https://rocketchat.atlassian.net/browse/CONN-434)


[CONN-434]: https://rocketchat.atlassian.net/browse/CONN-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
